tests/batching/test_requests.py

```diff
@@ -0,0 +1,372 @@
+import pytest
+
+from typing import List
+
+from mlserver.types import (
+    InferenceRequest,
+    RequestInput,
+    ResponseOutput,
+    InferenceResponse,
+    Parameters,
+)
+from mlserver.batching.requests import BatchedRequests
+
+
+@pytest.mark.parametrize(
+    "request_inputs,expected",
+    [
+        (
+            [
+                RequestInput(
+                    name="foo",
+                    datatype="INT32",
+                    shape=[1, 3],
+                    data=[1, 2, 3],
+                    parameters=Parameters(content_type="np"),
```

Comment:
hmm, what will happen if the content_type is different for different RequestInput, i.e. if one request is sent for some reason as numpy representation and one as dataframe representation. I guess batch happens before we apply the codec? I dont know actually what will happen.